libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -355,6 +355,51 @@ proc handleSubscribe(g: GossipSub, peer: PubSubPeer, topic: string, subscribe: b
 
   trace "gossip peers", peers = g.gossipsub.peers(topic), topic
 
+proc sendIWantMessages(g: GossipSub, peer: PubSubPeer, messages: seq[Message]) =
+  ## Send IWANT reply messages to a peer, splitting them into batches
+  ## to avoid exceeding maxMessageSize limits
+
+  if messages.len == 0:
+    return
+
+  trace "sending iwant reply messages", peer, count = messages.len
+
+  # Split messages into batches to avoid exceeding maxMessageSize
+  # Use 90% of maxMessageSize to account for protobuf overhead
+  let maxSafeSize = (g.maxMessageSize * 90) div 100
```

Comment:
Where does 90% come from? Is this in a spec somewhere? Could we get away with 95% or more?

Reply:
The question here is how this affects performance. I believe @richard-ramos and @ufarooqstatus can provide more information.